flexmeasures/data/services/sensors.py

```diff
@@ -328,3 +331,35 @@ def build_asset_jobs_data(
             )
 
     return jobs_data
+
+
+@lru_cache()
+def _get_sensor_stats(sensor: Sensor, ttl_hash=None):
+    return db.session.execute(
+        sa.select(
+            DataSource.name,
+            sa.func.min(TimedBelief.event_start),
+            sa.func.max(TimedBelief.event_start),
+            sa.func.min(TimedBelief.event_value),
+            sa.func.max(TimedBelief.event_value),
+            sa.func.avg(TimedBelief.event_value),
+            sa.func.sum(TimedBelief.event_value),
+            sa.func.count(TimedBelief.event_value),
+        )
+        .select_from(TimedBelief)
+        .join(DataSource, DataSource.id == TimedBelief.source_id)
+        .filter(TimedBelief.sensor_id == sensor.id)
+        .group_by(DataSource.name)
+    ).fetchall()
+
+
+def _get_ttl_hash(seconds=300):
+    """Return the same value withing seconds time period
+    Is needed to make LRU cache a TTL one.
+    """
+    return round(time.time() / seconds)
+
+
+def get_sensor_stats(sensor: Sensor):
```

Comment:
Type hint for return type is helpful.

In any case, I would prefer a dictionary, with data sources as keys. Given the `group_by`, that seems a good fit.